src/solvers/dg_p4est/containers_2d.jl

```diff
@@ -61,37 +69,82 @@ function calc_node_coordinates!(node_coordinates,
 end
 
 
-# Iterate over all interfaces and extract inner interface data to interface container
-# This function will be passed to p4est in init_interfaces! below
-function init_interfaces_iter_face(info, user_data)
-  if info.sides.elem_count != 2
-    # Not an inner interface
-    return nothing
-  end
-
-  sides = (unsafe_load_sc(p4est_iter_face_side_t, info.sides, 1),
-           unsafe_load_sc(p4est_iter_face_side_t, info.sides, 2))
+# A helper struct used in initialization methods below
+mutable struct InitSurfacesIterFaceUserData{Interfaces, Mortars, Boundaries, Mesh}
+  interfaces  ::Interfaces
+  interface_id::Int
+  mortars     ::Mortars
+  mortar_id   ::Int
+  boundaries  ::Boundaries
+  boundary_id ::Int
+  mesh        ::Mesh
+end
```

Comment:
I don't have a strong opinion on this. `Boundaries` is just a local name used within these ten lines and I would be fine with calling it `B` if you strongly prefer that. What do you think, @sloede?

Reply:
I think both styles are fine. Since the approach of naming the throw-away parametric type after the variable name is the [prevalent](https://github.com/trixi-framework/Trixi.jl/blob/f2972d95bcf7c8107dcbc93d83e41d509757a9dd/src/semidiscretization/semidiscretization_hyperbolic.jl#L10-L21) [style](https://github.com/trixi-framework/Trixi.jl/blob/f2972d95bcf7c8107dcbc93d83e41d509757a9dd/src/callbacks_step/amr.jl#L12-L18) [in](https://github.com/trixi-framework/Trixi.jl/blob/f2972d95bcf7c8107dcbc93d83e41d509757a9dd/src/solvers/dg_tree/dg.jl#L143-L146) [Trixi](https://github.com/trixi-framework/Trixi.jl/blob/f2972d95bcf7c8107dcbc93d83e41d509757a9dd/src/callbacks_step/visualization.jl#L3-L9), I suggest to stick to it since this is what most people working longer with Trixi expect to see.